psutil/_psutil_linux.c

```diff
@@ -541,14 +541,15 @@ psutil_net_if_stats(PyObject* self, PyObject* args) {
     close(sock);
     py_retlist = Py_BuildValue("[Oiii]", py_is_up, duplex, speed, mtu);
     if (!py_retlist)
-        goto error;
+        goto error_after_close;
     Py_DECREF(py_is_up);
     return py_retlist;
 
 error:
-    Py_XDECREF(py_is_up);
-    if (sock != 0)
+    if (sock != -1)
         close(sock);
+error_after_close:
+    Py_XDECREF(py_is_up);
```

Comment:
I kinda don't like having two separate goto handlers. I think `close()`sing the socket twice will not be a problem,